cmake/modules/AddSwiftUnittests.cmake

```diff
@@ -43,8 +43,17 @@ function(add_swift_unittest test_dirname)
     set_property(TARGET "${test_dirname}" APPEND_STRING PROPERTY
       LINK_FLAGS " -Xlinker -rpath -Xlinker ${SWIFT_LIBRARY_OUTPUT_INTDIR}/swift/macosx")
   elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
```

Comment:
Why use `CMAKE_SYSTEM_NAME` and not `SWIFT_HOST_VARIANT` here? The two seem to be inconsistently used in some places.

Reply:
`CMAKE_SYSTEM_NAME` is the name of the system we are targeting.  That should be preferred to enable cross-compilation.  The `SWIFT_HOST_VARIANT` is the target for the runtime components IIRC, but, we should be aiming to reduce that usage (as it prevents cross-compilation).

Reply:
No, the docs say SWIFT_HOST_VARIANT is the "Deployment OS for Swift host tools (the compiler)" and it is used that way in a [few](https://github.com/apple/swift/blob/1dc288fb2314fcaeb129473eb51c5824db1d2664/tools/CMakeLists.txt#L52) [places](https://github.com/apple/swift/blob/4b9f8464e44934f3bb0eef290c5aacc757b13655/unittests/runtime/CMakeLists.txt#L49). However, counting it up now, I see CMAKE_SYSTEM_NAME used much more in this repo.

A decision probably needs to be made on whether SWIFT_HOST_VARIANT should be defined at all, as right now its basically just an alias for CMAKE_SYSTEM_NAME. This pull actually tries to use SWIFT_HOST_VARIANT though, since CMAKE_HOST_SYSTEM_NAME/CMAKE_SYSTEM_NAME are set to "Linux" by default on Android.

Reply:
First off, I'm still not sure if you understand exactly what I'm doing here, [take a look at my latest long comment answering @drodriguez's questions](https://github.com/apple/swift/pull/23208#issuecomment-471817196). That CMake manual link you give is irrelevant because it's about cross-compiling with the NDK, which I'm not using here.

As for using CMAKE_SYSTEM_NAME instead to cross-compile the host tools, it is a possibility. However, someone, presumably gribozavr in 6670bb76ecc, chose to use SWIFT_HOST_VARIANT_SDK/SWIFT_HOST_VARIANT, presumably to give this project more flexibility in how it names and deals with the host OS in CMake.

I'm simply following that precedent by using SWIFT_HOST_VARIANT more. For example, build-script always sets the SWIFT_HOST_VARIANT* variables and never CMAKE_SYSTEM_NAME.

However, both are being used all over this project's CMake config now, so it will break if you actually try to cross-compile the host tools for another OS someday, unless you set _both_ CMAKE_SYSTEM_NAME and SWIFT_HOST_VARIANT*. :smile:

Reply:
@drodriguez,
> I looked a little to setting CMAKE_SYSTEM_NAME to Android, and it will work for cross-compiling with the NDK, but it might not work with the Termux environment.

Can you elaborate a bit more on what you did? What did you set the SWIFT_HOST_VARIANT_SDK and SWIFT_HOST_VARIANT to when you set the CMAKE_SYSTEM_NAME to Android? Given that this project doesn't use the CMake Android toolchain that @compnerd linked and rolls its own NDK support, I suspect the two will clash and not actually work. Maybe CMake didn't error out, but I suspect it would at least fail when building.

>It supposes that you are using the NDK, so the search paths might not work for what they want to do.

Depending on what you set the SWIFT_HOST_VARIANT* variables to and the aforementioned potential conflicts with CMake's built-in Android toolchain config, I suspect it wouldn't work for you with the NDK either.

>This is why I’m asking to try to use the NDK itself inside Termux. It will make the changes minimal.

I think they're already minimal enough. Take a look at my latest commit, which makes it easy to tell when the NDK is being used. Since you understably care much more about the NDK paths being easy to follow, I think the latest iteration of this pull does that.